### PR TITLE
IGNITE-20496 Enable Logit log storage by default

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-20496

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java [hunk 1]
```diff
@@ -97,6 +100,9 @@
  * Raft server implementation on top of forked JRaft library.
  */
 public class JraftServerImpl implements RaftServer {
+    /** Enables logit log storage. {@code true} by default. */
+    public static final String LOGIT_STORAGE_ENABLED_PROPERTY = "LOGIT_STORAGE_ENABLED";
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java:104]: Is using a system property a temporary solution?

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/option/StoreOptions.java [hunk 2]
```diff
@@ -27,13 +27,13 @@ public class StoreOptions {
 
     private static final String storagePath                   = "localLog";
 
-    // Default is 0.5G
-    private int                 segmentFileSize               = 1024 * 1024 * 512;
+    // Default is 64Mb.
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/option/StoreOptions.java:30]: There is probably a reason for the default to differ from the original default. Could it be put in this comment?

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/db/SegmentLogDB.java [hunk 3]
```diff
@@ -17,15 +17,15 @@
 
 package org.apache.ignite.raft.jraft.storage.logit.storage.db;
 
-import org.apache.ignite.raft.jraft.storage.logit.storage.file.FileType;
+import java.util.concurrent.ScheduledExecutorService;import org.apache.ignite.raft.jraft.storage.logit.storage.file.FileType;
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/db/SegmentLogDB.java:20]: Imports are on the same line

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/index/IndexFile.java [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 import org.apache.ignite.internal.logger.IgniteLogger;
 import org.apache.ignite.internal.logger.Loggers;
-import org.apache.ignite.raft.jraft.storage.logit.storage.file.AbstractFile;
+import org.apache.ignite.raft.jraft.option.RaftOptions;import org.apache.ignite.raft.jraft.storage.logit.storage.file.AbstractFile;
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/index/IndexFile.java:25]: Double imports

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java [hunk 5]
```diff
@@ -24,7 +24,7 @@
 import org.apache.ignite.internal.logger.Loggers;
 import org.apache.ignite.raft.jraft.entity.LogEntry;
 import org.apache.ignite.raft.jraft.entity.codec.v1.LogEntryV1CodecFactory;
-import org.apache.ignite.raft.jraft.storage.logit.storage.file.AbstractFile;
+import org.apache.ignite.raft.jraft.option.RaftOptions;import org.apache.ignite.raft.jraft.storage.logit.storage.file.AbstractFile;
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java:27]: Double imports

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java [hunk 6]
```diff
@@ -102,9 +102,9 @@ public byte[] lookupData(final long logIndex, final int pos) {
                     getFilePath(), logIndex, pos, this.header.getFirstLogIndex(), getLastLogIndex());
                 return null;
             }
-            if (pos > getFlushedPosition()) {
+            if (pos > getWrotePosition()) {
```
Comment (rpuch) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java:105]: I don't understand this change, could you please elaborate?

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java (hunk 1 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java:104]: Yes, it is temporary, by the time we release the product, only one log storage will be left. I'll say that in the comment

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/option/StoreOptions.java (hunk 2 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/option/StoreOptions.java:30]: I made it equal to defaults in Ignite 2. I'll mention that in the comment as well

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java (hunk 6 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/logit/storage/file/segment/SegmentFile.java:105]: I forgot to comment on it, I apologize. It is a bug fix, to my knowledge

**Review by rpuch — APPROVED** [review]

**Pull request merged**